### PR TITLE
restructure dirs

### PR DESCRIPTION
**Pull request by mjaskula**
- Moves the contrib dir into the software folder.
- completes the removal of the flash_nuke file and doc. I found that with the removal of flash_nuke, this readme wasn't relevant anymore. It was all covered better by programming_instructions.
- adds some make targets to make copying easier

I didn't touch the instructions on deploying europi.py to the pyboard, or deploying individual scripts. We can discuss in chat, but I'm not sure it's actually any easier now. I still think that this change is good for the other reasons (all software together, better testing setup)

Fixes issue #55

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,3 +5,12 @@ clean_docs:
 
 docs:
 	$(MAKE) -C docs html SPHINXOPTS="-W -n"
+
+clean:
+	find . -type d -name __pycache__ -print -exec rm -r {} \+
+
+deploy_firmware: clean
+	# requires rshell  https://github.com/dhylands/rshell
+	 rshell rsync software/firmware /pyboard/lib
```
Comment (awonak) [inline Makefile:14]: It looks like rsync doesn't support pattern matching, so this is going to copy the .MD files from contrib too. If we start to get close to the 2MB flash memory limit, we might want to use `cp` which does support matching, or look into a minification step.

https://github.com/dhylands/rshell#rsync
https://github.com/dhylands/rshell#cp

**Review by awonak — APPROVED** [review]
This looks good, I'm really liking the additional make target!

File: Makefile (hunk 1 above)
Comment (mjaskula) [reply Makefile:14]: I'm gonna do this now. It kinda makes us look sill to be copying files indiscriminately to the pico.

File: scripts/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+set -e
+
+rshell cp ${1} /pyboard/main.py
+rshell 'repl ~ import machine ~ machine.soft_reset()~'
```
Comment (awonak) [inline scripts/deploy.sh:5]: LGTM, at some point I may add a flag to allow the script to run and attach to the rshell serial output for print debugging purposes.

Basically just running it like this:
```shell
rshell 'repl ~ import main'
```

**Comment by awonak** [comment]
LGTM

**Pull request merged**